msal/src/authority/java/com/microsoft/identity/client/AadAuthority.java

```diff
@@ -114,6 +119,17 @@ boolean existsInResolvedAuthorityCache(final String userPrincipalName) {
         return RESOLVED_AUTHORITY.containsKey(mAuthorityUrl.toString());
     }
 
+    private void populateCommonCloudDiscoveryMetadata(InstanceDiscoveryResponse response) throws JSONException {
+        if (null == response
+                || null == response.getResponseClaims()
+                || response.getResponseClaims().isEmpty()) {
+            return;
+        }
+        //Ideally, this logic would have gone in AadAuthority#addToResolvedAuthorityCache(String)
+        // but I'm not going to refactor this too much for now.
+        AzureActiveDirectory.initializeCloudMetadata(mAuthorityUrl.getHost(), response.getResponseClaims());
```

Comment:
@shoatman Do you think it's fine that I'm using this Map for metadata tracking in the MSAL logic? Or would it be more appropriate to essentially duplicate it for MSAL?